lib/datadog/core/configuration/settings.rb

```diff
@@ -424,9 +424,29 @@ def initialize(*_)
             #
             # @default `DD_PROFILING_EXPERIMENTAL_TIMELINE_ENABLED` environment variable as a boolean, otherwise `false`
             option :experimental_timeline_enabled do |o|
+              o.after_set do |_, _, precedence|
+                unless precedence == Datadog::Core::Configuration::Option::Precedence::DEFAULT
+                  Datadog.logger.warn(
+                    'The profiling.advanced.experimental_timeline_enabled setting has been deprecated for removal ' \
+                    'and no longer does anything. Please remove it from your Datadog.configure block. ' \
+                    'The timeline feature counting is now controlled by the `timeline_enabled` setting instead.'
+                  )
+                end
+              end
+            end
+
+            # Controls data collection for the timeline feature.
+            #
+            # If you needed to disable this, please tell us why on <https://github.com/DataDog/dd-trace-rb/issues/new>,
+            # so we can fix it!
+            #
+            # TODO: In the future this setting may be deprecated and the timeline may no longer be optional.
```

Comment:
Minor: Is this TODO useful? It's a given that in the future anything may happen 😂

Reply:
You're right, it's a bit too much futurology. Removed in c862874ec8b0d94241bb2d801a191d339e755fe4 .